webviz_subsurface/plugins/_well_completions.py

```diff
@@ -0,0 +1,473 @@
+    https://github.com/equinor/webviz-subsurface-components/blob/master/inputSchema/wellCompletions.json
+    """
+
+    df = load_csv(ensemble_paths={ensemble: ensemble_path}, csv_file=compdat_file)
+    qc_compdat(df)
+    layer_zone_mapping = read_zone_layer_mapping(
+        ensemble_path=ensemble_path,
+        zone_layer_mapping_file=zone_layer_mapping_file,
+    )
+    well_attributes = read_well_attributes(
+        ensemble_path=ensemble_path,
+        well_attributes_file=well_attributes_file,
+    )
+
+    time_steps = sorted(df.DATE.unique())
+    realizations = np.asarray(sorted(df.REAL.unique()), dtype=np.int32)
+    layers = np.sort(df.K1.unique())
+
+    if layer_zone_mapping is None:
+        # use layers as zones
+        layer_zone_mapping = {layer: f"Layer{layer}" for layer in layers}
+
+    df["ZONE"] = df.K1.map(layer_zone_mapping)
+    zone_names = list(dict.fromkeys(layer_zone_mapping.values()))
+
+    result = {
+        "version": "1.0.0",
+        "stratigraphy": extract_stratigraphy(layer_zone_mapping, colors),
+        "timeSteps": time_steps,
+        "wells": extract_wells(
+            df, zone_names, time_steps, realizations, well_attributes
+        ),
+    }
+
+    return io.BytesIO(json.dumps(result).encode())
+
+
+def qc_compdat(compdat: pd.DataFrame) -> None:
+    """
+    QCs that the compdat data has the required format
+    """
+    needed_columns = ["WELL", "K1", "OP/SH", "KH"]
+    for column in needed_columns:
+        if column not in compdat:
+            raise ValueError(
+                f"Column {column} not found in compdat dataframe."
+                "This should not occur unless there has been changes to ecl2df."
+            )
+
+
+def get_time_series(df: pd.DataFrame, time_steps: list) -> tuple:
+    """
+    Create two lists with values for each time step
+    * the first one is on the form [0,0,0,1,1,1,1,-1,-1,-1] where '0' means no event,\
+    '1' is open, '-1' is shut.
+    * the second is with sum of kh values for the open compdats in each zone
+
+    The input data frame is assumed to contain data for single well,
+    single zone and single realization.
+    """
+    if df.empty:
+        return [0] * len(time_steps), [0] * len(time_steps)
+
+    events, kh_values = [], []
+    event_value, kh_value = 0, 0
+
+    for timestep in time_steps:
+        if timestep in df.DATE.unique():
+            df_timestep = df[df.DATE == timestep]
+            df_timestep_open = df_timestep[df_timestep["OP/SH"] == "OPEN"]
+
+            # if minimum one of the compdats for the zone is OPEN then the zone is considered open
+            event_value = 1 if not df_timestep_open.empty else -1
+            kh_value = df_timestep_open.KH.sum()
+
+        events.append(event_value)
+        kh_values.append(kh_value)
+    return events, kh_values
+
+
+def get_completion_events_and_kh(
+    df: pd.DataFrame, zone_names: list, time_steps: list
+) -> tuple:
+    # pylint: disable=unused-variable
```

Comment:
Instead of this pylint exception, you can probably mark the variable as unused by calling it `_`.